hyperspy/io_plugins/fei.py

```diff
@@ -503,17 +532,22 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
                             'Dim-%i_CalibrationOffset' % (i + 1)][0],
                         'scale': header[
                             'Dim-%i_CalibrationDelta' % (i + 1)][0],
-                        'units': header['Dim-%i_Units' % (i + 1)][0],
+                        # for image stack, the UnitsLength is 0 (no units)
+                        'units': header['Dim-%i_Units' % (i + 1)][0] if header['Dim-%i_UnitsLength'%(i+1)]>0 else 'a.u.',
                         'size': header['Dim-%i_DimensionSize' % (i + 1)][0],
                     })
                 array_shape.append(header['Dim-%i_DimensionSize' % (i + 1)][0])
+        units = "Unknown"
+        if objects is not None:
+            objects_dict = convert_xml_to_dict(objects[0])
+            units = guess_units_from_mode(objects_dict, header)
```

Comment:
Then, if we cannot guarantee that the xml chunck belongs to a given ser file, guessing the units in this way won't work, isn't it? If that's the case, I suggest to comment out these lines and explain in a comment that, once we find out how to match xml chunks to ser files using the UUID, this code can be re-enabled.